shared/test/unit/mocks/mock_language_list.js

```diff
@@ -11,8 +11,8 @@
         callback(langs);
       }
     },
-    get: function(callback) {
-      callback(langs, 'en-US');
+    get: function() {
+      Promise.resolve([langs, 'en-US']);
```

Comment:
You should also be able to change `_languages` above to a real Promise with `Promise.resolve`.  Right now it pretends to be a `then`-able but is in fact synchronous for easier testing.